datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/PatchEntityResolver.java

```diff
@@ -0,0 +1,159 @@
+import com.linkedin.datahub.graphql.resolvers.mutate.util.PatchResolverUtils;
+import com.linkedin.entity.client.EntityClient;
+import com.linkedin.metadata.authorization.PoliciesConfig;
+import com.linkedin.metadata.entity.EntityService;
+import com.linkedin.metadata.entity.IngestResult;
+import com.linkedin.metadata.models.AspectSpec;
+import com.linkedin.metadata.models.registry.EntityRegistry;
+import com.linkedin.metadata.utils.AuditStampUtils;
+import com.linkedin.mxe.GenericAspect;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.util.concurrent.CompletableFuture;
+import javax.annotation.Nonnull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RequiredArgsConstructor
+public class PatchEntityResolver implements DataFetcher<CompletableFuture<PatchEntityResult>> {
+
+  private final EntityService _entityService;
+  private final EntityClient _entityClient;
+  private final EntityRegistry _entityRegistry;
+
+  @Override
+  public CompletableFuture<PatchEntityResult> get(DataFetchingEnvironment environment)
+      throws Exception {
+    final PatchEntityInput input =
+        bindArgument(environment.getArgument("input"), PatchEntityInput.class);
+    final QueryContext context = environment.getContext();
+
+    return GraphQLConcurrencyUtils.supplyAsync(
+        () -> {
+          try {
+            return patchEntity(input, context);
+          } catch (Exception e) {
+            log.error("Failed to patch entity: {}", e.getMessage(), e);
+            // Use a placeholder URN if none was provided
+            String urn = input.getUrn() != null ? input.getUrn() : "urn:li:unknown:error";
+            return new PatchEntityResult(urn, null, false, e.getMessage());
+          }
+        },
+        this.getClass().getSimpleName(),
+        "patchEntity");
+  }
+
+  private PatchEntityResult patchEntity(
+      @Nonnull PatchEntityInput input, @Nonnull QueryContext context) throws Exception {
+
+    // Handle URN resolution - either provided or auto-generated
+    final Urn entityUrn =
+        PatchResolverUtils.resolveEntityUrn(input.getUrn(), input.getEntityType());
+    final Authentication authentication = context.getAuthentication();
+
+    // Check authorization using GraphQL authorization pattern
+    if (!isAuthorized(input, context)) {
+      throw new AuthorizationException(
+          authentication.getActor().toUrnStr() + " is unauthorized to update entities.");
+    }
+
+    // Validate aspect exists
+    final AspectSpec aspectSpec =
+        _entityRegistry
+            .getEntitySpec(entityUrn.getEntityType())
+            .getAspectSpec(input.getAspectName());
+    if (aspectSpec == null) {
+      throw new IllegalArgumentException(
+          "Aspect "
+              + input.getAspectName()
+              + " not found for entity type "
+              + entityUrn.getEntityType());
+    }
```

Comment:
there's some common functionality between this resolver and the PatchEntitiesResolver that we could combine. i kind of expected this mostly to all be pushed out into a utils class

that utils class could create MCPs and the resolver could simply ingest and return the expected result

so then both this resolver and the bulk one basically call `createPatchEntitiesMcps` or something and this resolver passes in a list of 1

basically looking to minimize the duplication between the two